app/handlers/sessions_create.rb

```diff
@@ -57,7 +57,14 @@ def handle
         status = :authentication_taken
       else
         return outputs[:status] = :new_signin_required if user_signin_is_too_old?
-        run(TransferAuthentications, authentication, current_user)
+        auth_transfer = run(TransferAuthentications, authentication, current_user)
+
+        if auth_transfer.outputs[:same_provider]
```

Comment:
1. Very true! Will fix with 3.

2. The problem is that if I use a `fatal_error`, then the bug "Accounts explodes..." will remain in existence. So we really don't want to error out just because a user is trying to use a different account with the same provider they've already authenticated with. That's why I must output a status. Plus, if I do return a status, then we'll be able to handle it gracefully in `sessions#create`—with a nicer [alert](https://github.com/openstax/accounts/pull/345/files#diff-d5241d488259f32ecbe2f636133e5ddaR89)—which is all we want for now.

3., 4., 5. Will do! Ok.